### PR TITLE
Group versions

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: app/src/test/java/com/github/bsideup/liiklus/GroupVersionTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+        assertThat(getRecords(Optional.of(groupVersion)).blockFirst(Duration.ofSeconds(10)).getOffset())
+                .isEqualTo(nextOffset);
+
+        // Old style should also work
+        assertThat(getRecords(Optional.empty()).blockFirst(Duration.ofSeconds(10)).getOffset())
+                .isEqualTo(nextOffset);
+    }
+
+    @Test
```
Comment (mrnakumar) [inline app/src/test/java/com/github/bsideup/liiklus/GroupVersionTest.java]: should we also add check when a higher version 'ack' and lower version tries to getRecords? I think version will not be decreasing, but just to add.

File: api/src/main/java/com/github/bsideup/liiklus/positions/GroupId.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        Matcher matcher = VERSION_PATTERN.matcher(str);
+
+        if (matcher.matches()) {
+            MatchResult result = matcher.toMatchResult();
+
+            return GroupId.of(
+                    result.group(1),
+                    Optional.ofNullable(result.group(2)).map(Integer::parseInt)
+            );
+        } else {
```
Comment (lanwen) [inline api/src/main/java/com/github/bsideup/liiklus/positions/GroupId.java]: you can omit this else

Comment (bsideup) [reply api/src/main/java/com/github/bsideup/liiklus/positions/GroupId.java]: AFAIK `result.group(2)` will fail

Comment (lanwen) [reply api/src/main/java/com/github/bsideup/liiklus/positions/GroupId.java]: I mean operator, simply return

Comment (bsideup) [reply api/src/main/java/com/github/bsideup/liiklus/positions/GroupId.java]: ah, ok, I get it

**Pull request merged**